aead/src/lib.rs

```diff
@@ -72,22 +68,31 @@ impl fmt::Display for Error {
 impl core::error::Error for Error {}
 
 /// Nonce: single-use value for ensuring ciphertexts are unique
-pub type Nonce<A> = Array<u8, <A as AeadCore>::NonceSize>;
+pub type Nonce<A> = Array<u8, <A as Aead>::NonceSize>;
 
 /// Tag: authentication code which ensures ciphertexts are authentic
-pub type Tag<A> = Array<u8, <A as AeadCore>::TagSize>;
+pub type Tag<A> = Array<u8, <A as Aead>::TagSize>;
+
+/// Enum which specifies tag position used by an AEAD algorithm.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum TagKind {
```

Comment:
I would suggest:

```suggestion
pub enum TagPosition {
```

...which more clearly describes what this enum is for

Reply:
Renamed.